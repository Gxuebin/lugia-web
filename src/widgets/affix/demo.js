/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Affix from './affix';
import Button from '../button';

export default class AffixDemo extends React.Component {
  constructor() {
    super();
    this.EleRef = null;
    this.Ele2Ref = null;
    this.state = {
      display: true,
    };
  }
  render() {
    const { display } = this.state;
    return (
      <div>
        {display ? (
          <div style={{ width: '200px', height: '1200px' }}>
            <div style={{ width: '200px', height: '200px' }} />
            <Affix offsetTop={50}>
              <Button>affix-top</Button>
            </Affix>

            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.EleRef = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '100px' }} />
                <Affix offsetTop={50} target={() => this.EleRef}>
                  <Button>affix-top</Button>
                </Affix>
              </div>
            </div>
            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.Ele2Ref = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '200px' }} />
                <Affix offsetBottom={50} target={() => this.Ele2Ref}>
                  <Button>affix2-bottom</Button>
                </Affix>
              </div>
            </div>

            <div style={{ width: '200px', height: '500px' }} />
            <Affix offsetBottom={50}>
              <Button>affix-bottom</Button>
            </Affix>
            <div style={{ width: '200px', height: '500px' }} />
          </div>
        ) : (
          <div style={{ width: '200px', height: '1200px' }} />
        )}
      </div>
    );
  }
}
