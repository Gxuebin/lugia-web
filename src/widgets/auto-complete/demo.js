import * as React from 'react';
import AutoComplete from './';

const data = [
  'Nikcy Romero',
  'Armin van Buuren',
  'Hardwell',
  'Zedd',
  'Kazze',

  'Vicetone',
  'Martin Garrix',
  'David Guetta',
  'The Chainsmokers',
  'Kygo',

  'Axwell ^ Ingrosso',
  'Dimitri Vegas & Like Mike',
  'Calvin Harris',
  'Avicci',
  'Fedde Le Grand',

  'Tiesto',
  'Snoop Dogg',
  'Bassjackers',
  'Sebastian Ingrosso',
  'Swedish House Mafia',
  'Alesso',
  'Afrojack',
  'Knife Party',
  'Dannic',
  'R3hab',
];

export default class AutoCompleteBounded extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      menuData: data,
    };
  }
  static getDerivedStateFromProps(props: any, state: any) {
    const hasValueInProps = 'value' in props;
    const value = hasValueInProps ? props.value : state ? state.value : props.defaultValue;
    if (!state) {
      return {
        value,
      };
    }
    return {
      value,
    };
  }

  render() {
    const { menuData, value } = this.state;
    return <AutoComplete showOldValue value={value} data={menuData} onChange={this.onChange} />;
  }

  onChange = (value: string) => {
    this.search(value);
    this.setState({ value });
  };

  search(query: string) {
    let menuData;
    let rowSet = [];
    const len = data.length;

    for (let i = 0; i < len; i++) {
      const row = data[i];
      if (this.searchValue(query, row)) {
        rowSet.push(row);
      }
      if (query === row) {
        rowSet = [];
        break;
      }
    }

    if (rowSet.length === len) {
      menuData = data;
    } else {
      menuData = rowSet;
    }
    this.setState({ menuData });
  }

  searchValue = (query: string, row: string): boolean => {
    return row.indexOf(query) !== -1;
  };
}

// export default class Demo extends React.Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       menuData: data,
//       value: '',
//     };
//   }
//   render() {
//     return (
//       <AutoComplete
//         data={this.state.menuData}
//         onChange={this.onChange}
//         placeholder={'请输入'}
//         showOldValue={true}
//       />
//     );
//   }

//   onChange = value => {
//     this.search(value);
//   };

//   search(query) {
//     let menuData;
//     let rowSet = [];
//     const len = data.length;

//     for (let i = 0; i < len; i++) {
//       const row = data[i];
//       if (query === row) {
//         rowSet = [];
//         break;
//       }

//       if (this.searchValue(query, row)) {
//         rowSet.push(row);
//       }
//     }

//     if (rowSet.length === len) {
//       menuData = data;
//     } else {
//       menuData = rowSet;
//     }
//     this.setState({ menuData });
//   }

//   searchValue = (query, row) => {
//     return row.indexOf(query) !== -1 || row === query;
//   };
// }
