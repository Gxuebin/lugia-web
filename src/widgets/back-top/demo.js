/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import BackTop from './back-top';
import Widget from '../consts/index';
import Theme from '../theme';

const Demo = styled.div`
  & > span > div > div {
    bottom: ${props => 50 + props.cur * 50}px;
  }
`;
const DemoBack = styled.div`
  width: 40px;
  height: 40px;
  line-height: 40px;
  text-align: center;
  border-radius: 4px;
  border: 1px solid #e8e8e8;
`;
export default class AffixDemo extends React.Component<any, any> {
  EleRef: any;
  render() {
    const view = {
      [Widget.BackTop]: {
        normal: {
          color: 'red',
          background: { backgroundColor: 'orange' },
          width: 100,
          height: 100,
          borderRadius: 100,
          opacity: 0.7,
          border: {
            borderWidth: 1,
            borderStyle: 'solid',
            borderColor: 'red',
          },
        },
      },
    };
    return (
      <div style={{ height: '1600px' }}>
        <BackTop />
        <Demo cur={1}>
          <BackTop>
            <DemoBack>UP</DemoBack>
          </BackTop>
        </Demo>
        <Demo cur={2}>
          <Theme config={view}>
            <BackTop />
          </Theme>
        </Demo>
        <div
          style={{ width: '200px', height: '200px', overflowY: 'scroll', marginTop: '400px' }}
          ref={node => (this.EleRef = node)}
        >
          <Demo cur={3}>
            <div style={{ width: '20px', height: '300px' }} />
            <BackTop visibilityHeight={80} target={() => this.EleRef} />
          </Demo>
        </div>
      </div>
    );
  }
}
