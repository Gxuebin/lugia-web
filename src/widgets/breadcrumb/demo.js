import * as React from 'react';
import Breadcrumb from './index';
import styled from 'styled-components';
import Icon from '../icon';
import { HashRouter as Router, Route, Switch, Link, withRouter } from 'react-router-dom';

const Hr = styled.div`
  height: 2px;
  background: orange;
  margin: 20px 0;
`;

const CommonIcon = styled(Icon)`
  color: #000;
  border-radius: 50%;
`;

const routes = [
  {
    path: 'index',
    breadcrumbName: '首页',
  },
  {
    path: 'first',
    breadcrumbName: '一级面包屑',
  },
  {
    path: 'second',
    breadcrumbName: '二级面包屑',
  },
  {
    path: 'third',
    breadcrumbName: '当前页面',
  },
];

export default class Demo extends React.Component<any, any> {
  render() {
    return (
      <div>
        {/*Breadcrumb 和 Breadcrumb.Item配合使用，使用href属性时，可跳转 */}
        <Breadcrumb separator={'>'}>
          <Breadcrumb.Item href="">主页</Breadcrumb.Item>
          <Breadcrumb.Item href="">一级菜单</Breadcrumb.Item>
          <Breadcrumb.Item href="">二级菜单</Breadcrumb.Item>
          <Breadcrumb.Item>三级菜单</Breadcrumb.Item>
        </Breadcrumb>
        <Hr />
        {/* 传入routes路由属性，使用默认的 renderItem函数,生成面包屑组件 */}
        <Breadcrumb separator={'>'} routes={routes} />
        <Hr />
        {/* 传入Icon图标 */}
        <Breadcrumb>
          <Breadcrumb.Item href="">
            <CommonIcon iconClass="lugia-icon-logo_chrome" type="home" />
            <span>一级面包屑菜单 </span>
          </Breadcrumb.Item>
          <Breadcrumb.Item href="">
            <span>二级面包屑菜单</span>
            <CommonIcon iconClass="lugia-icon-financial_smile" type="user" />
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <span>二级面包屑菜单</span>
            <CommonIcon iconClass="lugia-icon-logo_apple" type="user" />
          </Breadcrumb.Item>
        </Breadcrumb>
      </div>
    );
  }
}

// //       demo4
// // !!!!!!!!!!!!!!!!!!!这个组件这么使用有bug，不懂什么原因
// // 使用的自定义设置的 renderItem回调函数,传入routes路由属性，生成面包屑组件=====================
// import * as React from 'react';
// import Breadcrumb from './index';
// import { Link, Router } from 'react-router-dom';

// const routes = [
//   {
//     path: 'index',
//     breadcrumbName: '首页',
//   },
//   {
//     path: 'first',
//     breadcrumbName: '一级面包屑',
//   },
//   {
//     path: 'second',
//     breadcrumbName: '二级面包屑',
//   },
//   {
//     path: 'third',
//     breadcrumbName: '当前页面',
//   },
// ];

// function renderItem(route, params, routes, paths) {
//   const last = routes.indexOf(route) === routes.length - 1;
//   return last ? (
//     <span>{route.breadcrumbName}</span>
//   ) : (
//     <Link to={paths.join('/')}>{route.breadcrumbName}</Link>
//   );
// }
// export default class Demo extends React.Component<any, any> {
//   render() {
//     return (
//       <Router>
//         <Breadcrumb renderItem={renderItem} routes={routes} />
//       </Router>
//     );
//   }
// }

//     demo6
//  给路由传入params 参数
// import * as React from 'react';
// import Breadcrumb from './index';
// const routes = [
//   {
//     name: 'home',
//     breadcrumbName: 'Home',
//     path: '/',
//   },
//   {
//     name: 'apps',
//     breadcrumbName: 'Application List',
//     path: 'apps',
//   },
//   {
//     name: 'app',
//     breadcrumbName: 'Application:id',
//     path: ':id',
//   },
//   {
//     name: 'detail',
//     breadcrumbName: 'Detail',
//     path: 'detail',
//   },
// ];

// export default class Demo extends React.Component<any, any> {
//   render() {
//     return <Breadcrumb params={{ id: 1, class: 2 }} routes={routes} />;
//   }
// }
