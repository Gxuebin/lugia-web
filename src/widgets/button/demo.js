/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import Button from './index';

export default () => {
  return <Button type="danger">hello</Button>;
};
