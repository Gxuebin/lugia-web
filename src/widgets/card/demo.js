/**
 *
 * create by liangguodong on 2018/11/29
 *
 * @flow
 */
import React from 'react';
import Card from './index';
import styled from 'styled-components';
import Theme from '../theme';
import Tabs from '../tabs/tabs';
import Widget from '../consts';

const Wrapper = styled.div`
  margin-left: 50px;
  margin-top: 50px;
  display: inline-block;
`;
const CardContainer = styled.div`
  text-align: center;
  width: 100%;
  height: 100%;
`;
const Price = styled.div`
  text-align: center;
  font-size: 16px;
  color: #666;
  margin-bottom: 26px;
  width: 100%;
`;
const PriceNum = styled.div`
  text-align: center;
  color: #4d63ff;
  display: inline-block;
`;
const Buy = styled.button`
  position: absolute;
  text-align: center;
  font-size: 14px;
  border-radius: 20px;
  margin-top: 32px;
  color: #fff;
  bottom: 10%;
  width: 80%;
  background-color: #4d63ff;
  height: 32px;
  border: 2px solid #4d63ff;
  left: 10%;
`;

class AmountCard extends React.Component<Object, Object> {
  render() {
    const price = '9.88';
    let int = 0;
    let decimals = 0;
    if (price.indexOf('.') > 0) {
      int = price.substring(0, price.indexOf('.'));
      decimals = price.substring(price.indexOf('.'));
    }
    return (
      <CardContainer>
        <Price>{'价格'}</Price>
        <PriceNum>
          <PriceNum style={{ fontSize: 32 }}>{int}</PriceNum>
          <PriceNum style={{ fontSize: 18 }}>{decimals}</PriceNum>
        </PriceNum>
        <Buy>{'购买'}</Buy>
      </CardContainer>
    );
  }
}

export default () => {
  const view = {
    [Widget.Card]: {
      width: 300,
      height: 200,
    },
  };
  const combo = {
    register: {
      width: 700,
      height: 300,
    },
    vertical: {
      width: 200,
      height: 220,
    },
  };
  const avatar = {
    [Widget.Avatar]: {
      width: 80,
      height: 80,
    },
  };
  const cardImage = {
    [Widget.CardImage]: {
      width: 200,
      height: 130,
    },
  };
  const defaultData = [
    {
      icon: 'lugia-icon-financial_archive',
      title: 1111111111,
      content: 1111,
    },
    {
      icon: 'lugia-icon-financial_archive',
      title: 22222222222,
      content: 22222,
    },
    {
      icon: 'lugia-icon-financial_archive',
      title: 3333333,
      content: 333,
    },
    {
      icon: 'lugia-icon-financial_archive',
      title: 44444444,
      content: 4444444,
    },
    {
      icon: 'lugia-icon-financial_archive',
      title: 55555555,
      content: 5555555,
    },
  ];
  return (
    <Wrapper>
      <p>基本样式</p>
      <Wrapper>
        <Card
          title={'this is title'}
          description={'this is description'}
          shadow={'always'}
          operation={'操作'}
        />
      </Wrapper>
      <p>基本样式</p>
      <Theme config={view}>
        <Wrapper>
          <Card
            title={<div>{'this is title'}</div>}
            description={[
              <div>{'this is description'}</div>,
              <div>{'this is description'}</div>,
              <div>{'this is description'}</div>,
              <div>{'this is description'}</div>,
            ]}
            shadow={'always'}
          />
        </Wrapper>
      </Theme>
      <p>children样式 </p>
      <Wrapper>
        <Card>
          <div>children 样式的卡片</div>
        </Card>
      </Wrapper>
      <p>头像样式</p>
      <Theme config={avatar}>
        <Wrapper>
          <Card
            type={'avatar'}
            title={'this is title'}
            description={'this is description'}
            avatar={
              'https://gss1.bdstatic.com/-vo3dSag_xI4khGkpoWK1HF6hhy/baike/c0%3Dbaike92%2C5%2C5%2C92%2C30/sign=7878a9471d38534398c28f73f27adb1b/738b4710b912c8fc8e9cace6f1039245d68821a9.jpg'
            }
            shadow={'hover'}
          />
        </Wrapper>
      </Theme>
      <Theme config={avatar}>
        <p>头像样式</p>
        <Wrapper>
          <Card
            type={'avatar'}
            title={'this is title'}
            description={'this is description'}
            imageOrientation={'vertical'}
            avatar={
              'https://gss1.bdstatic.com/-vo3dSag_xI4khGkpoWK1HF6hhy/baike/c0%3Dbaike92%2C5%2C5%2C92%2C30/sign=7878a9471d38534398c28f73f27adb1b/738b4710b912c8fc8e9cace6f1039245d68821a9.jpg'
            }
            shadow={'hover'}
          />
        </Wrapper>
      </Theme>
      <p>图片样式</p>
      <Wrapper>
        <Card
          type={'image'}
          title={'this is title'}
          imageOrientation={'horizontal'}
          description={'this is description'}
          image={
            'https://gss1.bdstatic.com/-vo3dSag_xI4khGkpoWK1HF6hhy/baike/c0%3Dbaike92%2C5%2C5%2C92%2C30/sign=4f88e0c6b3de9c82b268f1dd0de8eb6f/f9198618367adab4973d1fbc8bd4b31c8601e464.jpg'
          }
          shadow={'hover'}
        />
      </Wrapper>
      <Theme config={cardImage}>
        <p>图片样式</p>
        <Wrapper>
          <Card
            viewClass={'cardImage'}
            type={'image'}
            imageOrientation={'vertical'}
            title={'this is title'}
            description={'this is description'}
            image={
              'https://gss1.bdstatic.com/-vo3dSag_xI4khGkpoWK1HF6hhy/baike/c0%3Dbaike92%2C5%2C5%2C92%2C30/sign=4f88e0c6b3de9c82b268f1dd0de8eb6f/f9198618367adab4973d1fbc8bd4b31c8601e464.jpg'
            }
            shadow={'hover'}
          />
        </Wrapper>
      </Theme>
      <p>组合样式</p>
      <Theme config={combo}>
        <Wrapper>
          <Card
            operation={'操作'}
            viewClass={'register'}
            type={'combo'}
            content={<Tabs data={defaultData} />}
            shadow={'hover'}
          />
        </Wrapper>

        <p>组合样式</p>
        <Wrapper>
          <Card viewClass={'vertical'} type={'combo'} content={<AmountCard />} shadow={'hover'} />
        </Wrapper>
      </Theme>
    </Wrapper>
  );
};
