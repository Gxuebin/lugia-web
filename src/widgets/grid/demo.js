/**
 *
 * create by guorg
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';
import Grid from './';
const { Row, Col } = Grid;
const Wrapper = styled.div`
  & > div > div {
    text-align: center;
  }
  & > div > div:nth-child(odd) {
    background: rgba(0, 160, 233, 0.7);
  }

  & > div > div:nth-child(even) {
    background: #00a0e9;
  }
`;
export const ColDemo = () => {
  return (
    <div>
      <Col span={6}>col-6</Col>
      <br />
      <Col span={6} offset={6}>
        col-6 col-offset-6
      </Col>
      <br />
      <Col span={6} push={6}>
        col-6 col-push-6
      </Col>
      <br />
      <Col span={6} pull={6}>
        col-6 col-pull-6
      </Col>
      <br />
      <Col span={6} order={1}>
        col-6 col-order-1
      </Col>
      <br />
    </div>
  );
};
export const RowDemo = () => {
  const data = [
    { span: 5, children: 'col-6' },
    { span: 5, children: 'col-6' },
    { span: 5, children: 'col-6' },
    { span: 5, children: 'col-6' },
  ];
  return (
    <div>
      <Wrapper>
        <Row>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row>
          <Col span={6}>col-6</Col>
          {123}
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row>
          <Col span={6} offset={6}>
            col-6 col-offset-6
          </Col>
          <Col span={6} offset={6}>
            col-6 col-offset-6
          </Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row>
          <Col span={18} push={6}>
            col-18 col-push-6
          </Col>
          <Col span={6} pull={18}>
            col-6 col-pull-18
          </Col>
        </Row>
      </Wrapper>
      <br />
      <p>type: flex, justify</p>
      <Wrapper>
        <Row type="flex" justify="start">
          <Col span={4}>col-4 justify-start</Col>
          <Col span={4}>col-4 justify-start</Col>
          <Col span={4}>col-4 justify-start</Col>
          <Col span={4}>col-4 justify-start</Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row type="flex" justify="center">
          <Col span={4}>col-4 justify-center</Col>
          <Col span={4}>col-4 justify-center</Col>
          <Col span={4}>col-4 justify-center</Col>
          <Col span={4}>col-4 justify-center</Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row type="flex" justify="end">
          <Col span={4}>col-4 justify-end</Col>
          <Col span={4}>col-4 justify-end</Col>
          <Col span={4}>col-4 justify-end</Col>
          <Col span={4}>col-4 justify-end</Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row type="flex" justify="spaceBetween">
          <Col span={4}>col-4 justify-spaceBetween</Col>
          <Col span={4}>col-4 justify-spaceBetween</Col>
          <Col span={4}>col-4 justify-spaceBetween</Col>
          <Col span={4}>col-4 justify-spaceBetween</Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row type="flex" justify="spaceAround">
          <Col span={4}>col-4 justify-spacAround</Col>
          <Col span={4}>col-4 justify-spacAround</Col>
          <Col span={4}>col-4 justify-spacAround</Col>
          <Col span={4}>col-4 justify-spacAround</Col>
        </Row>
      </Wrapper>
      <br />
      <p>type: flex, align</p>
      <Wrapper>
        <Row type="flex" justify="start" align="top">
          <Col span={4}>
            <div style={{ height: 80 }}>col-4 align-top</div>
          </Col>
          <Col span={4}>
            <div style={{ height: 60 }}>col-4 align-top</div>
          </Col>
          <Col span={4}>
            <div style={{ height: 100 }}>col-4 align-top</div>
          </Col>
          <Col span={4}>
            <div style={{ height: 120 }}>col-4 align-top</div>
          </Col>
        </Row>
      </Wrapper>
      <br />
      <Wrapper>
        <Row type="flex" justify="center" align="middle">
          <Col span={4}>
            <div style={{ height: 80 }}>col-4 align-middle</div>
          </Col>
          <Col span={4}>
            <div style={{ height: 60 }}>col-4 align-middle</div>
          </Col>
          <Col span={4}>
            <div style={{ height: 100 }}>col-4 align-middle</div>
          </Col>
          <Col span={4}>
            <div style={{ height: 120 }}>col-4 align-middle</div>
          </Col>
        </Row>
      </Wrapper>
      <Wrapper>
        <Row gutter={{ xs: 8, sm: 16, md: 24, xl: 24, lg: 24 }} data={data} />
      </Wrapper>
    </div>
  );
};
