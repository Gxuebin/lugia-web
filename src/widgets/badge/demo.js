/**
 *
 * create by liangguodong on 2018/8/27
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Badge from './';
import { Col, Row } from '../grid';
import Widget from '../consts';
import Theme from '../theme';
import Icon from '../icon';

const Box = styled.div`
  width: 40px;
  height: 40px;
  background: #ccc;
  margin-left: 10px;
`;
const Wrapper = styled.div`
  margin-left: 10px;
  margin-top: 50px;
`;

class Turn extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  click = type => () => {
    const count = type === 'plus' ? this.state.count + 1 : this.state.count - 1;
    this.setState({ count });
  };

  render() {
    const config = {
      [Widget.Badge]: {
        children: {
          [Widget.NumberTurn]: {
            normal: { position: { top: -5, right: -5 } },
          },
        },
      },
    };
    return (
      <Wrapper>
        <Theme config={config}>
          <Row>
            <Col span={4}>
              <Badge count={this.state.count} showZero overflowCount={99}>
                <Box />
              </Badge>
              <Icon
                style={{ fontSize: '2em' }}
                iconClass="lugia-icon-reminder_plus_square_o"
                onClick={this.click('plus')}
              />
              <Icon
                style={{ fontSize: '2em' }}
                iconClass="lugia-icon-reminder_minus_square_o"
                onClick={this.click('minus')}
              />
            </Col>
          </Row>
        </Theme>
      </Wrapper>
    );
  }
}

export default () => {
  const view = {
    [Widget.Badge]: {
      normal: {
        position: { top: -5, right: 0 },
        background: { backgroundColor: 'orange' },
        color: 'black',
      },
    },
  };
  return (
    <div>
      <Theme
        config={{
          [Widget.Badge]: {
            children: {
              [Widget.NumberTurn]: {
                normal: { position: { top: -5, right: -5 } },
              },
            },
          },
        }}
      >
        <Wrapper>
          <Row>
            <Col span={4}>
              <Badge count={4}>
                <Box />
              </Badge>
            </Col>
            <Col span={4}>
              <Badge count={99}>
                <Box />
              </Badge>
            </Col>
            <Col span={4}>
              <Badge>
                <Box />
              </Badge>
            </Col>
          </Row>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <Row>
              <Col span={4}>
                <Badge count={0}>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <Badge showZero>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <Badge showZero count={0}>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <a href="www.baidu.com">
                  <Badge>
                    <Box />
                  </Badge>
                </a>
              </Col>
            </Row>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Row>
            <Col span={4}>
              <Theme
                config={{
                  green: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'green' },
                    },
                  },
                }}
              >
                <Badge viewClass="green">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  purple: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'purple' },
                    },
                  },
                }}
              >
                <Badge viewClass="purple">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  yellow: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'yellow' },
                    },
                  },
                }}
              >
                <Badge viewClass="yellow">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  blue: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'blue' },
                    },
                  },
                }}
              >
                <Badge viewClass="blue">
                  <Box />
                </Badge>
              </Theme>
            </Col>
          </Row>
        </Wrapper>
        <Turn />
      </Theme>
    </div>
  );
};
