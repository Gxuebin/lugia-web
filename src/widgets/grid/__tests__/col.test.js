/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import chai from 'chai';

import 'jest-styled-components';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ColDemo } from '../demo';
import renderer from 'react-test-renderer';

const { expect: exp } = chai;

Enzyme.configure({ adapter: new Adapter() });

describe('Col', () => {
  beforeEach(() => {});

  it('css', () => {
    const Target = <ColDemo />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
});
