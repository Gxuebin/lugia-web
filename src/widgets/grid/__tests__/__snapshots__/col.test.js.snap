// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col css 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  float: left;
  position: relative;
  width: 25%;
  margin-left: 0%;
}

.c1 {
  display: block;
  box-sizing: border-box;
  float: left;
  position: relative;
  width: 25%;
  margin-left: 25%;
}

.c2 {
  display: block;
  box-sizing: border-box;
  float: left;
  position: relative;
  width: 25%;
  margin-left: 0%;
  left: 25%;
}

.c3 {
  display: block;
  box-sizing: border-box;
  float: left;
  position: relative;
  width: 25%;
  margin-left: 0%;
  right: 25%;
}

.c4 {
  display: block;
  box-sizing: border-box;
  float: left;
  position: relative;
  width: 25%;
  margin-left: 0%;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

<div>
  <div
    className="c0"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    span={6}
  >
    col-6
  </div>
  <br />
  <div
    className="c1"
    offset={6}
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    span={6}
  >
    col-6 col-offset-6
  </div>
  <br />
  <div
    className="c2"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    span={6}
  >
    col-6 col-push-6
  </div>
  <br />
  <div
    className="c3"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    span={6}
  >
    col-6 col-pull-6
  </div>
  <br />
  <div
    className="c4"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    order={1}
    span={6}
  >
    col-6 col-order-1
  </div>
  <br />
</div>
`;
